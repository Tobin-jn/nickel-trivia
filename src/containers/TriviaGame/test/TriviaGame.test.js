/* eslint-disable */

import React from "react";
import { shallow } from "enzyme";
import { TriviaGame, mapStateToProps, mapDispatchToProps } from "../index";
import { mockState, mockQuestion } from './mocks';

describe("TriviaGame", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TriviaGame 
      currentQuestion={mockQuestion}
      gameCount={1}
      score={5}
    />);
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it("should return a currentQuestion object in the props object", () => {
    const expected = {        
      category: "Geography",
      question: "Which city is the capital of the United States of America?",
      correct_answer: "Washington D.C",
      answers: [],
      id: 1,
      asked: false,
    }

    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps.currentQuestion).toEqual(expected)
  })

  it('should return a number for gameCount in the props object', () => {
    const expected = 2

    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps.gameCount).toEqual(expected)
  })

  it('should return a number for score in the props object', () => {
    const expected = 5

    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps.score).toEqual(expected)
  })
})

// describe('mapDispatchToProps', () => {
//   it('should call dispatch with...', () => {

//   })
// })