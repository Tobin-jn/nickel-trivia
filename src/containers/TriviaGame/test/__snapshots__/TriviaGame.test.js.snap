// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaGame should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TriviaGame
    currentQuestion={
      Object {
        "answers": Array [],
        "asked": false,
        "category": "Geography",
        "correct_answer": "Washington D.C",
        "id": 1,
        "question": "Which city is the capital of the United States of America?",
      }
    }
    gameCount={1}
    score={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Game!
        </h1>,
        <Connect(Categories) />,
        <Connect(Question) />,
        <div
          className="score"
        >
          <h1
            className="score-text"
          >
            Score:
          </h1>
          <h4
            className="score-number"
          >
            5
          </h4>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Game!",
        },
        "ref": null,
        "rendered": "Game!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="score-text"
            >
              Score:
            </h1>,
            <h4
              className="score-number"
            >
              5
            </h4>,
          ],
          "className": "score",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Score:",
              "className": "score-text",
            },
            "ref": null,
            "rendered": "Score:",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "score-number",
            },
            "ref": null,
            "rendered": 5,
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Game!
          </h1>,
          <Connect(Categories) />,
          <Connect(Question) />,
          <div
            className="score"
          >
            <h1
              className="score-text"
            >
              Score:
            </h1>
            <h4
              className="score-number"
            >
              5
            </h4>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Game!",
          },
          "ref": null,
          "rendered": "Game!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="score-text"
              >
                Score:
              </h1>,
              <h4
                className="score-number"
              >
                5
              </h4>,
            ],
            "className": "score",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Score:",
                "className": "score-text",
              },
              "ref": null,
              "rendered": "Score:",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
                "className": "score-number",
              },
              "ref": null,
              "rendered": 5,
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
